googleapi/googleapi.go

```diff
@@ -211,6 +211,20 @@ type MediaOption interface {
 	setOptions(o *MediaOptions)
 }
 
+type bufferOption []byte
+
+func (bp bufferOption) setOptions(o *MediaOptions) {
+	o.Buffer = bp
+}
+
+// WithBuffer returns MediaOption which sets buffer used for media uploads.
+// It needs to be at least MinUploadChunkSize long.
+// If used together with ChunkSize, buffer needs to be at least that long.
+// If not set, each upload will allocate its own memory.
```

Comment:
I wonder if it is worth calling out explicitly about re-using buffers only after requests complete. If someone passes in the same buffer to two concurrent calls it will lead to a data race.

Reply:
Added mention about data race.